diff --git a/src/gz_ros2_control_plugin.cpp b/src/gz_ros2_control_plugin.cpp
index 7952e78..3464bf7 100644
--- a/src/gz_ros2_control_plugin.cpp
+++ b/src/gz_ros2_control_plugin.cpp
@@ -209,7 +209,7 @@ GazeboSimROS2ControlPluginPrivate::GetEnabledJoints(
         {
           RCLCPP_INFO(
             node_->get_logger(),
-            "[gz_ros2_control] Fixed joint [%s] (Entity=%lu)] is skipped",
+            "[gz_ros2_control] Fixed joint [%s] (Entity=%llu)] is skipped",
             jointName.c_str(), jointEntity);
           continue;
         }
@@ -220,7 +220,7 @@ GazeboSimROS2ControlPluginPrivate::GetEnabledJoints(
         {
           RCLCPP_WARN(
             node_->get_logger(),
-            "[gz_ros2_control] Joint [%s] (Entity=%lu)] is of unsupported type."
+            "[gz_ros2_control] Joint [%s] (Entity=%llu)] is of unsupported type."
             " Only joints with a single axis are supported.",
             jointName.c_str(), jointEntity);
           continue;
@@ -229,7 +229,7 @@ GazeboSimROS2ControlPluginPrivate::GetEnabledJoints(
         {
           RCLCPP_WARN(
             node_->get_logger(),
-            "[gz_ros2_control] Joint [%s] (Entity=%lu)] is of unknown type",
+            "[gz_ros2_control] Joint [%s] (Entity=%llu)] is of unknown type",
             jointName.c_str(), jointEntity);
           continue;
         }
@@ -271,7 +271,7 @@ void GazeboSimROS2ControlPlugin::Configure(
   if (!model.Valid(_ecm)) {
     RCLCPP_ERROR(
       logger,
-      "[Gazebo ROS 2 Control] Failed to initialize because [%s] (Entity=%lu)] is not a model."
+      "[Gazebo ROS 2 Control] Failed to initialize because [%s] (Entity=%llu)] is not a model."
       "Please make sure that Gazebo ROS 2 Control is attached to a valid model.",
       model.Name(_ecm).c_str(), _entity);
     return;
diff --git a/src/gz_system.cpp b/src/gz_system.cpp
index f99ece9..17d659c 100644
--- a/src/gz_system.cpp
+++ b/src/gz_system.cpp
@@ -205,7 +205,7 @@ bool GazeboSimSystem::initSim(
     this->nh_->get_logger(), "hold_joints (system): " << this->dataPtr->hold_joints_ << std::endl);
 
 
-  RCLCPP_DEBUG(this->nh_->get_logger(), "n_dof_ %lu", this->dataPtr->n_dof_);
+  RCLCPP_DEBUG(this->nh_->get_logger(), "n_dof_ %zu", this->dataPtr->n_dof_);
 
   this->dataPtr->joints_.resize(this->dataPtr->n_dof_);
 
