diff --git a/src/gz_system.cpp b/src/gz_system.cpp
index f99ece9..17d659c 100644
--- a/src/gz_system.cpp
+++ b/src/gz_system.cpp
@@ -205,7 +205,7 @@ bool GazeboSimSystem::initSim(
     this->nh_->get_logger(), "hold_joints (system): " << this->dataPtr->hold_joints_ << std::endl);
 
 
-  RCLCPP_DEBUG(this->nh_->get_logger(), "n_dof_ %lu", this->dataPtr->n_dof_);
+  RCLCPP_DEBUG(this->nh_->get_logger(), "n_dof_ %zu", this->dataPtr->n_dof_);
 
   this->dataPtr->joints_.resize(this->dataPtr->n_dof_);
 
