diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f2b40688..fe7df25ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,7 +80,10 @@ add_library(mppi_critics SHARED
   src/critics/twirling_critic.cpp
   src/critics/velocity_deadband_critic.cpp
 )
-target_compile_options(mppi_critics PUBLIC -fconcepts -O3)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  target_compile_options(mppi_critics PUBLIC -fconcepts)
+endif()
+target_compile_options(mppi_critics PUBLIC -O3)
 target_include_directories(mppi_critics
   PUBLIC
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
