diff --git a/include/rmw/types.h b/include/rmw/types.h
index 2258e25..96ac5d1 100644
--- a/include/rmw/types.h
+++ b/include/rmw/types.h
@@ -213,6 +213,9 @@ typedef struct RMW_PUBLIC_TYPE rmw_subscription_s
 
   /// Indicates whether content filtered topic of this subscription is enabled
   bool is_cft_enabled;
+
+  /// Indicates whether this subscription supports content filtered topic feature
+  bool is_cft_supported;
 } rmw_subscription_t;
 
 /// A handle to an rmw service
