diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca30488f7..48c38bcf7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ find_package(kdl_parser REQUIRED)
 find_package(moveit_msgs REQUIRED)
 # Enforce system version liboctomap-dev
 # https://github.com/moveit/moveit2/issues/2862
-find_package(octomap 1.9.7...<1.10.0 REQUIRED)
+find_package(octomap REQUIRED)
 find_package(octomap_msgs REQUIRED)
 find_package(osqp REQUIRED)
 find_package(pluginlib REQUIRED)
@@ -43,6 +43,16 @@ find_package(visualization_msgs REQUIRED)
 
 # Finds Boost Components
 include(ConfigExtras.cmake)
+set(MOVEIT_BOOST_TARGETS
+    Boost::chrono
+    Boost::date_time
+    Boost::filesystem
+    Boost::iostreams
+    Boost::program_options
+    Boost::regex
+    Boost::serialization
+    Boost::system
+    Boost::thread)
 
 add_subdirectory(collision_detection_bullet)
 add_subdirectory(collision_detection_fcl)
diff --git a/collision_detection/CMakeLists.txt b/collision_detection/CMakeLists.txt
index e3f8fd212..001f0e3db 100644
--- a/collision_detection/CMakeLists.txt
+++ b/collision_detection/CMakeLists.txt
@@ -16,20 +16,6 @@ target_include_directories(
 include(GenerateExportHeader)
 generate_export_header(moveit_collision_detection)
 
-ament_target_dependencies(
-  moveit_collision_detection
-  eigen_stl_containers
-  pluginlib
-  rclcpp
-  rmw_implementation
-  urdf
-  urdfdom
-  urdfdom_headers
-  srdfdom
-  visualization_msgs
-  tf2_eigen
-  geometric_shapes
-  octomap)
 target_include_directories(
   moveit_collision_detection BEFORE
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
@@ -41,7 +27,18 @@ set_target_properties(moveit_collision_detection
                       PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
 
 target_link_libraries(moveit_collision_detection moveit_robot_state
-                      moveit_utils)
+                      moveit_utils
+                      ${eigen_stl_containers_TARGETS}
+                      pluginlib::pluginlib
+                      rclcpp::rclcpp
+                      ${rmw_implementation_TARGETS}
+                      urdf::urdf
+                      urdfdom_headers::urdfdom_headers
+                      ${srdfdom_TARGETS}
+                      ${visualization_msgs_TARGETS}
+                      tf2_eigen::tf2_eigen
+                      ${geometric_shapes_TARGETS}
+                      octomap)
 
 # unit tests
 if(BUILD_TESTING)
diff --git a/collision_detection/include/moveit/collision_detection/collision_common.hpp b/collision_detection/include/moveit/collision_detection/collision_common.hpp
index 19777a044..af3a89442 100644
--- a/collision_detection/include/moveit/collision_detection/collision_common.hpp
+++ b/collision_detection/include/moveit/collision_detection/collision_common.hpp
@@ -141,7 +141,33 @@ struct CostSource
   }
 };
 
-struct CollisionResult;
+/** \brief Representation of a collision checking result */
+struct CollisionResult
+{
+  using ContactMap = std::map<std::pair<std::string, std::string>, std::vector<Contact> >;
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  /** \brief Clear a previously stored result */
+  void clear()
+  {
+    collision = false;
+    distance = std::numeric_limits<double>::max();
+    contact_count = 0;
+    contacts.clear();
+    cost_sources.clear();
+  }
+  /** \brief Throttled warning printing the first collision pair, if any. All collisions are logged at DEBUG level */
+  void print() const;
+  /** \brief True if collision was found, false otherwise */
+  bool collision = false;
+  /** \brief Closest distance between two bodies */
+  double distance = std::numeric_limits<double>::max();
+  /** \brief Number of contacts returned */
+  std::size_t contact_count = 0;
+  /** \brief A map returning the pairs of body ids in contact, plus their contact details */
+  ContactMap contacts;
+  /** \brief These are the individual cost sources when costs are computed */
+  std::set<CostSource> cost_sources;
+};
 
 /** \brief Representation of a collision checking request */
 struct CollisionRequest
@@ -159,9 +185,6 @@ struct CollisionRequest
   /** \brief If true, compute proximity distance */
   bool distance = false;
 
-  /** \brief If true, return detailed distance information. Distance must be set to true as well */
-  bool detailed_distance = false;
-
   /** \brief If true, a collision cost is computed */
   bool cost = false;
 
@@ -330,42 +353,4 @@ struct DistanceResult
   }
 };
 
-/** \brief Representation of a collision checking result */
-struct CollisionResult
-{
-  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
-
-  /** \brief Clear a previously stored result */
-  void clear()
-  {
-    collision = false;
-    distance = std::numeric_limits<double>::max();
-    distance_result.clear();
-    contact_count = 0;
-    contacts.clear();
-    cost_sources.clear();
-  }
-
-  /** \brief Throttled warning printing the first collision pair, if any. All collisions are logged at DEBUG level */
-  void print() const;
-
-  /** \brief True if collision was found, false otherwise */
-  bool collision = false;
-
-  /** \brief Closest distance between two bodies */
-  double distance = std::numeric_limits<double>::max();
-
-  /** \brief Distance data for each link */
-  DistanceResult distance_result;
-
-  /** \brief Number of contacts returned */
-  std::size_t contact_count = 0;
-
-  /** \brief A map returning the pairs of body ids in contact, plus their contact details */
-  using ContactMap = std::map<std::pair<std::string, std::string>, std::vector<Contact> >;
-  ContactMap contacts;
-
-  /** \brief These are the individual cost sources when costs are computed */
-  std::set<CostSource> cost_sources;
-};
 }  // namespace collision_detection
diff --git a/collision_detection_bullet/CMakeLists.txt b/collision_detection_bullet/CMakeLists.txt
index feadeb653..2ecd227da 100644
--- a/collision_detection_bullet/CMakeLists.txt
+++ b/collision_detection_bullet/CMakeLists.txt
@@ -18,35 +18,37 @@ target_include_directories(
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 set_target_properties(moveit_collision_detection_bullet
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_collision_detection_bullet SYSTEM BULLET)
-ament_target_dependencies(
+target_include_directories(moveit_collision_detection_bullet SYSTEM
+                           PUBLIC ${BULLET_INCLUDE_DIRS})
+target_link_libraries(
   moveit_collision_detection_bullet
-  rclcpp
-  rmw_implementation
-  urdf
-  urdfdom
-  urdfdom_headers
-  visualization_msgs
-  octomap_msgs)
-target_link_libraries(moveit_collision_detection_bullet
-                      moveit_collision_detection moveit_utils)
+  PUBLIC ${BULLET_LIBRARIES}
+         rclcpp::rclcpp
+         ${rmw_implementation_TARGETS}
+         urdf::urdf
+         urdfdom_headers::urdfdom_headers
+         ${visualization_msgs_TARGETS}
+         ${octomap_msgs_TARGETS}
+  PRIVATE moveit_collision_detection moveit_utils)
 
 add_library(collision_detector_bullet_plugin SHARED
             src/collision_detector_bullet_plugin_loader.cpp)
 set_target_properties(collision_detector_bullet_plugin
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(collision_detector_bullet_plugin SYSTEM BULLET)
-ament_target_dependencies(
-  collision_detector_bullet_plugin
-  rclcpp
-  urdf
-  visualization_msgs
-  pluginlib
-  rmw_implementation
-  octomap_msgs)
+target_include_directories(collision_detector_bullet_plugin SYSTEM
+                           PRIVATE ${BULLET_INCLUDE_DIRS})
 target_link_libraries(
-  collision_detector_bullet_plugin moveit_collision_detection_bullet
-  moveit_planning_scene moveit_utils)
+  collision_detector_bullet_plugin
+  PRIVATE ${BULLET_LIBRARIES}
+          rclcpp::rclcpp
+          urdf::urdf
+          ${visualization_msgs_TARGETS}
+          pluginlib::pluginlib
+          ${rmw_implementation_TARGETS}
+          ${octomap_msgs_TARGETS}
+          moveit_collision_detection_bullet
+          moveit_planning_scene
+          moveit_utils)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(
diff --git a/collision_detection_fcl/CMakeLists.txt b/collision_detection_fcl/CMakeLists.txt
index ece406480..2c2a9e4b8 100644
--- a/collision_detection_fcl/CMakeLists.txt
+++ b/collision_detection_fcl/CMakeLists.txt
@@ -11,26 +11,29 @@ target_include_directories(
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 set_target_properties(moveit_collision_detection_fcl
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
+target_link_libraries(
   moveit_collision_detection_fcl
-  rclcpp
-  rmw_implementation
-  urdf
-  urdfdom
-  urdfdom_headers
-  visualization_msgs)
-target_link_libraries(moveit_collision_detection_fcl moveit_collision_detection
-                      moveit_utils fcl)
+  PUBLIC rclcpp::rclcpp
+         ${rmw_implementation_TARGETS}
+         urdf::urdf
+         urdfdom_headers::urdfdom_headers
+         ${visualization_msgs_TARGETS}
+  PRIVATE moveit_collision_detection moveit_utils fcl)
 
 add_library(collision_detector_fcl_plugin SHARED
             src/collision_detector_fcl_plugin_loader.cpp)
 set_target_properties(collision_detector_fcl_plugin
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(collision_detector_fcl_plugin rclcpp urdf
-                          visualization_msgs pluginlib rmw_implementation)
 target_link_libraries(
-  collision_detector_fcl_plugin moveit_collision_detection_fcl
-  moveit_planning_scene moveit_utils)
+  collision_detector_fcl_plugin
+  PRIVATE rclcpp::rclcpp
+          urdf::urdf
+          ${visualization_msgs_TARGETS}
+          pluginlib::pluginlib
+          ${rmw_implementation_TARGETS}
+          moveit_collision_detection_fcl
+          moveit_planning_scene
+          moveit_utils)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(
diff --git a/collision_detection_fcl/src/collision_env_fcl.cpp b/collision_detection_fcl/src/collision_env_fcl.cpp
index b57e0767e..c655ed069 100644
--- a/collision_detection_fcl/src/collision_env_fcl.cpp
+++ b/collision_detection_fcl/src/collision_env_fcl.cpp
@@ -290,10 +290,6 @@ void CollisionEnvFCL::checkSelfCollisionHelper(const CollisionRequest& req, Coll
     dreq.enableGroup(getRobotModel());
     distanceSelf(dreq, dres, state);
     res.distance = dres.minimum_distance.distance;
-    if (req.detailed_distance)
-    {
-      res.distance_result = dres;
-    }
   }
 }
 
@@ -347,10 +343,6 @@ void CollisionEnvFCL::checkRobotCollisionHelper(const CollisionRequest& req, Col
     dreq.enableGroup(getRobotModel());
     distanceRobot(dreq, dres, state);
     res.distance = dres.minimum_distance.distance;
-    if (req.detailed_distance)
-    {
-      res.distance_result = dres;
-    }
   }
 }
 
diff --git a/collision_distance_field/CMakeLists.txt b/collision_distance_field/CMakeLists.txt
index 63b63a013..978a816ca 100644
--- a/collision_distance_field/CMakeLists.txt
+++ b/collision_distance_field/CMakeLists.txt
@@ -15,12 +15,17 @@ target_include_directories(
 set_target_properties(moveit_collision_distance_field
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-ament_target_dependencies(moveit_collision_distance_field urdf
-                          visualization_msgs tf2_eigen geometric_shapes octomap)
-
 target_link_libraries(
-  moveit_collision_distance_field moveit_planning_scene moveit_distance_field
-  moveit_collision_detection moveit_robot_state)
+  moveit_collision_distance_field
+  PUBLIC urdf::urdf
+         ${visualization_msgs_TARGETS}
+         tf2_eigen::tf2_eigen
+         ${geometric_shapes_TARGETS}
+         octomap
+  PRIVATE moveit_planning_scene
+          moveit_distance_field
+          moveit_collision_detection
+          moveit_robot_state)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(
@@ -33,10 +38,12 @@ if(BUILD_TESTING)
 
   ament_add_gtest(test_collision_distance_field
                   test/test_collision_distance_field.cpp)
-  ament_target_dependencies(test_collision_distance_field geometric_shapes
-                            octomap srdfdom resource_retriever)
   target_link_libraries(
     test_collision_distance_field
+    ${geometric_shapes_TARGETS}
+    octomap
+    ${srdfdom_TARGETS}
+    ${resource_retriever_TARGETS}
     moveit_collision_distance_field
     moveit_collision_detection
     moveit_distance_field
diff --git a/constraint_samplers/CMakeLists.txt b/constraint_samplers/CMakeLists.txt
index 25cda7dc6..8cf0a3d6e 100644
--- a/constraint_samplers/CMakeLists.txt
+++ b/constraint_samplers/CMakeLists.txt
@@ -9,10 +9,11 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_constraint_samplers
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_constraint_samplers urdf urdfdom
-                          urdfdom_headers visualization_msgs)
 target_link_libraries(
   moveit_constraint_samplers
+  urdf::urdf
+  urdfdom_headers::urdfdom_headers
+  ${visualization_msgs_TARGETS}
   moveit_robot_trajectory
   moveit_robot_state
   moveit_kinematic_constraints
@@ -43,7 +44,7 @@ if(BUILD_TESTING)
                   test/pr2_arm_kinematics_plugin.cpp test/pr2_arm_ik.cpp)
   target_include_directories(test_constraint_samplers
                              PUBLIC ${geometry_msgs_INCLUDE_DIRS})
-  ament_target_dependencies(test_constraint_samplers kdl_parser)
-  target_link_libraries(test_constraint_samplers moveit_test_utils moveit_utils
+  target_link_libraries(test_constraint_samplers ${kdl_parser_TARGETS}
+                        moveit_test_utils moveit_utils
                         moveit_constraint_samplers)
 endif()
diff --git a/distance_field/CMakeLists.txt b/distance_field/CMakeLists.txt
index 165d82a7a..daa8c8511 100644
--- a/distance_field/CMakeLists.txt
+++ b/distance_field/CMakeLists.txt
@@ -9,16 +9,11 @@ target_include_directories(
 target_link_libraries(moveit_distance_field moveit_macros moveit_utils)
 set_target_properties(moveit_distance_field
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
-  moveit_distance_field
-  Boost
-  eigen_stl_containers
-  urdfdom
-  urdfdom_headers
-  visualization_msgs
-  geometric_shapes
-  tf2_eigen
-  octomap)
+target_link_libraries(moveit_distance_field ${MOVEIT_BOOST_TARGETS}
+                      ${eigen_stl_containers_TARGETS}
+                      urdfdom_headers::urdfdom_headers
+                      ${visualization_msgs_TARGETS}
+                      ${geometric_shapes_TARGETS} tf2_eigen::tf2_eigen octomap)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/dynamics_solver/CMakeLists.txt b/dynamics_solver/CMakeLists.txt
index 04afe66ef..87f95ae22 100644
--- a/dynamics_solver/CMakeLists.txt
+++ b/dynamics_solver/CMakeLists.txt
@@ -6,8 +6,10 @@ target_include_directories(
 set_target_properties(moveit_dynamics_solver
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-ament_target_dependencies(moveit_dynamics_solver urdf urdfdom_headers
-                          orocos_kdl visualization_msgs kdl_parser)
 target_link_libraries(moveit_dynamics_solver moveit_robot_state moveit_utils)
+target_link_libraries(
+  moveit_dynamics_solver urdf::urdf urdfdom_headers::urdfdom_headers
+  ${orocos_kdl_LIBRARIES} ${visualization_msgs_TARGETS} ${kdl_parser_TARGETS})
+target_include_directories(moveit_dynamics_solver PUBLIC ${orocos_kdl_INCLUDE_DIRS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/exceptions/CMakeLists.txt b/exceptions/CMakeLists.txt
index 26591d955..b638c4bad 100644
--- a/exceptions/CMakeLists.txt
+++ b/exceptions/CMakeLists.txt
@@ -5,8 +5,7 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_exceptions PROPERTIES VERSION
                                                    "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_exceptions Boost rclcpp urdfdom
-                          urdfdom_headers)
-target_link_libraries(moveit_exceptions moveit_utils)
+target_link_libraries(moveit_exceptions moveit_utils ${MOVEIT_BOOST_TARGETS}
+                      rclcpp::rclcpp urdfdom_headers::urdfdom_headers)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/kinematic_constraints/CMakeLists.txt b/kinematic_constraints/CMakeLists.txt
index 51f33e495..079f3e173 100644
--- a/kinematic_constraints/CMakeLists.txt
+++ b/kinematic_constraints/CMakeLists.txt
@@ -3,7 +3,7 @@ if(WIN32)
   # "$<TARGET_FILE_DIR:${PROJECT_NAME}>;$<TARGET_FILE_DIR:${PROJECT_NAME}_TestPlugins1>")
 else()
   set(APPEND_LIBRARY_DIRS
-      "${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_BINARY_DIR}/../utils;${CMAKE_CURRENT_BINARY_DIR}/../robot_model;${CMAKE_CURRENT_BINARY_DIR}/../robot_state;${CMAKE_CURRENT_BINARY_DIR}/../collision_detection_fcl"
+      "${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_BINARY_DIR}/../utils;${CMAKE_CURRENT_BINARY_DIR}/../robot_model;${CMAKE_CURRENT_BINARY_DIR}/../robot_state;${CMAKE_CURRENT_BINARY_DIR}/../collision_detection;${CMAKE_CURRENT_BINARY_DIR}/../collision_detection_fcl"
   )
 endif()
 
@@ -16,19 +16,12 @@ target_include_directories(
 set_target_properties(moveit_kinematic_constraints
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-ament_target_dependencies(
-  moveit_kinematic_constraints
-  urdf
-  urdfdom
-  urdfdom_headers
-  tf2_geometry_msgs
-  geometry_msgs
-  visualization_msgs
-  tf2_eigen)
-
 target_link_libraries(
-  moveit_kinematic_constraints moveit_collision_detection_fcl
-  moveit_kinematics_base moveit_robot_state moveit_robot_model moveit_utils)
+  moveit_kinematic_constraints moveit_collision_detection
+  moveit_collision_detection_fcl
+  moveit_kinematics_base moveit_robot_state moveit_robot_model moveit_utils
+  urdf::urdf urdfdom_headers::urdfdom_headers tf2_geometry_msgs::tf2_geometry_msgs
+  ${geometry_msgs_TARGETS} ${visualization_msgs_TARGETS} tf2_eigen::tf2_eigen)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/kinematics_base/CMakeLists.txt b/kinematics_base/CMakeLists.txt
index 180253900..3faedd50d 100644
--- a/kinematics_base/CMakeLists.txt
+++ b/kinematics_base/CMakeLists.txt
@@ -12,15 +12,10 @@ target_include_directories(
 )# for this library
 
 # This line ensures that messages are built before the library is built
-ament_target_dependencies(
-  moveit_kinematics_base
-  rclcpp
-  urdf
-  urdfdom_headers
-  srdfdom
-  moveit_msgs
-  geometric_shapes
-  geometry_msgs)
+target_link_libraries(
+  moveit_kinematics_base rclcpp::rclcpp urdf::urdf
+  urdfdom_headers::urdfdom_headers ${srdfdom_TARGETS} ${moveit_msgs_TARGETS}
+  ${geometric_shapes_TARGETS} ${geometry_msgs_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/moveit_kinematics_base_export.h
diff --git a/kinematics_metrics/CMakeLists.txt b/kinematics_metrics/CMakeLists.txt
index e760b834e..b4b034ee2 100644
--- a/kinematics_metrics/CMakeLists.txt
+++ b/kinematics_metrics/CMakeLists.txt
@@ -6,10 +6,9 @@ target_include_directories(
 set_target_properties(moveit_kinematics_metrics
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-ament_target_dependencies(moveit_kinematics_metrics urdf urdfdom_headers
-                          visualization_msgs)
-
 target_link_libraries(moveit_kinematics_metrics moveit_robot_model
-                      moveit_robot_state moveit_utils)
+                      moveit_robot_state moveit_utils urdf::urdf
+                      urdfdom_headers::urdfdom_headers
+                      ${visualization_msgs_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/online_signal_smoothing/CMakeLists.txt b/online_signal_smoothing/CMakeLists.txt
index 1f1912c6d..f5da9f7c0 100644
--- a/online_signal_smoothing/CMakeLists.txt
+++ b/online_signal_smoothing/CMakeLists.txt
@@ -10,7 +10,7 @@ include(GenerateExportHeader)
 generate_export_header(moveit_smoothing_base)
 set_target_properties(moveit_smoothing_base
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_smoothing_base rclcpp tf2_eigen)
+target_link_libraries(moveit_smoothing_base rclcpp::rclcpp tf2_eigen::tf2_eigen)
 
 # Plugin implementations
 add_library(moveit_acceleration_filter SHARED src/acceleration_filter.cpp)
@@ -24,11 +24,8 @@ generate_parameter_library(moveit_acceleration_filter_parameters
                            src/acceleration_filter_parameters.yaml)
 target_link_libraries(
   moveit_acceleration_filter moveit_acceleration_filter_parameters
-  moveit_robot_state moveit_smoothing_base osqp::osqp)
-ament_target_dependencies(
-  moveit_acceleration_filter srdfdom # include dependency from
-                                     # moveit_robot_model
-  pluginlib)
+  moveit_robot_state moveit_smoothing_base osqp::osqp ${srdfdom_TARGETS}
+  pluginlib::pluginlib)
 
 add_library(moveit_butterworth_filter SHARED src/butterworth_filter.cpp)
 generate_export_header(moveit_butterworth_filter)
@@ -41,11 +38,8 @@ generate_parameter_library(moveit_butterworth_filter_parameters
                            src/butterworth_parameters.yaml)
 target_link_libraries(
   moveit_butterworth_filter moveit_butterworth_filter_parameters
-  moveit_robot_model moveit_smoothing_base)
-ament_target_dependencies(
-  moveit_butterworth_filter
-  srdfdom # include dependency from moveit_robot_model
-  pluginlib)
+  moveit_robot_model moveit_smoothing_base ${srdfdom_TARGETS}
+  pluginlib::pluginlib)
 
 add_library(moveit_ruckig_filter SHARED src/ruckig_filter.cpp)
 generate_export_header(moveit_ruckig_filter)
@@ -57,10 +51,8 @@ generate_parameter_library(moveit_ruckig_filter_parameters
                            src/ruckig_filter_parameters.yaml)
 target_link_libraries(
   moveit_ruckig_filter moveit_robot_state moveit_ruckig_filter_parameters
-  moveit_smoothing_base ruckig::ruckig)
-ament_target_dependencies(
-  moveit_ruckig_filter srdfdom # include dependency from moveit_robot_model
-  pluginlib)
+  moveit_smoothing_base ruckig::ruckig ${srdfdom_TARGETS}
+  pluginlib::pluginlib)
 
 # Installation
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/planning_interface/CMakeLists.txt b/planning_interface/CMakeLists.txt
index 338765e2f..5604024a9 100644
--- a/planning_interface/CMakeLists.txt
+++ b/planning_interface/CMakeLists.txt
@@ -6,9 +6,14 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_planning_interface
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_planning_interface moveit_msgs urdf urdfdom
-                          urdfdom_headers)
-target_link_libraries(moveit_planning_interface moveit_robot_trajectory
-                      moveit_robot_state moveit_planning_scene moveit_utils)
+target_link_libraries(
+  moveit_planning_interface
+  PUBLIC ${moveit_msgs_TARGETS}
+         urdf::urdf
+         urdfdom_headers::urdfdom_headers
+  PRIVATE moveit_robot_trajectory
+          moveit_robot_state
+          moveit_planning_scene
+          moveit_utils)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/planning_scene/CMakeLists.txt b/planning_scene/CMakeLists.txt
index debb3e5b6..f083cfc34 100644
--- a/planning_scene/CMakeLists.txt
+++ b/planning_scene/CMakeLists.txt
@@ -10,17 +10,13 @@ target_include_directories(
 # TODO: Fix the versioning
 set_target_properties(moveit_planning_scene
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
-  moveit_planning_scene
-  Boost
-  rclcpp
-  urdfdom
-  urdfdom_headers
-  octomap_msgs
-  octomap)
-
 target_link_libraries(
   moveit_planning_scene
+  ${MOVEIT_BOOST_TARGETS}
+  rclcpp::rclcpp
+  urdfdom_headers::urdfdom_headers
+  ${octomap_msgs_TARGETS}
+  octomap
   moveit_robot_model
   moveit_robot_state
   moveit_exceptions
@@ -47,8 +43,8 @@ if(BUILD_TESTING)
 
   ament_add_gtest(test_planning_scene test/test_planning_scene.cpp
                   APPEND_LIBRARY_DIRS "${APPEND_LIBRARY_DIRS}")
-  ament_target_dependencies(test_planning_scene geometric_shapes srdfdom)
-  target_link_libraries(test_planning_scene moveit_test_utils
+  target_link_libraries(test_planning_scene ${geometric_shapes_TARGETS}
+                        ${srdfdom_TARGETS} moveit_test_utils
                         moveit_planning_scene)
 
   ament_add_gtest(test_collision_objects test/test_collision_objects.cpp
diff --git a/robot_model/CMakeLists.txt b/robot_model/CMakeLists.txt
index 2886ed4f0..0ab6358ab 100644
--- a/robot_model/CMakeLists.txt
+++ b/robot_model/CMakeLists.txt
@@ -28,25 +28,23 @@ target_include_directories(
     $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_robot_model
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
-  moveit_robot_model
-  angles
-  moveit_msgs
-  Eigen3
-  eigen_stl_containers
-  geometric_shapes
-  urdf
-  urdfdom_headers
-  srdfdom
-  visualization_msgs)
 target_link_libraries(moveit_robot_model moveit_exceptions moveit_macros
-                      moveit_utils)
+                      moveit_utils
+                      ${angles_TARGETS}
+                      ${moveit_msgs_TARGETS}
+                      Eigen3::Eigen
+                      ${eigen_stl_containers_TARGETS}
+                      ${geometric_shapes_TARGETS}
+                      urdf::urdf
+                      urdfdom_headers::urdfdom_headers
+                      ${srdfdom_TARGETS}
+                      ${visualization_msgs_TARGETS})
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
   ament_add_gtest(test_robot_model test/test.cpp)
-  ament_target_dependencies(test_robot_model rclcpp)
-  target_link_libraries(test_robot_model moveit_test_utils moveit_robot_model)
+  target_link_libraries(test_robot_model rclcpp::rclcpp moveit_test_utils
+                        moveit_robot_model)
 endif()
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/robot_state/CMakeLists.txt b/robot_state/CMakeLists.txt
index a906be706..101144a5c 100644
--- a/robot_state/CMakeLists.txt
+++ b/robot_state/CMakeLists.txt
@@ -7,16 +7,14 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_robot_state PROPERTIES VERSION
                                                     ${${PROJECT_NAME}_VERSION})
-ament_target_dependencies(
-  moveit_robot_state
-  eigen_stl_containers
-  urdf
-  tf2_geometry_msgs
-  geometric_shapes
-  urdfdom_headers
-  Boost)
 target_link_libraries(moveit_robot_state moveit_robot_model
-                      moveit_kinematics_base moveit_transforms)
+                      moveit_kinematics_base moveit_transforms
+                      ${eigen_stl_containers_TARGETS}
+                      urdf::urdf
+                      tf2_geometry_msgs::tf2_geometry_msgs
+                      ${geometric_shapes_TARGETS}
+                      urdfdom_headers::urdfdom_headers
+                      ${MOVEIT_BOOST_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
@@ -58,7 +56,7 @@ if(BUILD_TESTING)
 
   ament_add_google_benchmark(robot_state_benchmark
                              test/robot_state_benchmark.cpp)
-  ament_target_dependencies(robot_state_benchmark kdl_parser)
-  target_link_libraries(robot_state_benchmark moveit_robot_model
-                        moveit_test_utils moveit_robot_state)
+  target_link_libraries(robot_state_benchmark ${kdl_parser_TARGETS}
+                        moveit_robot_model moveit_test_utils
+                        moveit_robot_state)
 endif()
diff --git a/robot_trajectory/CMakeLists.txt b/robot_trajectory/CMakeLists.txt
index c4b26ebb8..a5729acf3 100644
--- a/robot_trajectory/CMakeLists.txt
+++ b/robot_trajectory/CMakeLists.txt
@@ -5,11 +5,10 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_robot_trajectory
                       PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
-ament_target_dependencies(moveit_robot_trajectory rclcpp urdfdom
-                          urdfdom_headers)
 
 target_link_libraries(moveit_robot_trajectory moveit_robot_model
-                      moveit_robot_state moveit_utils)
+                      moveit_robot_state moveit_utils rclcpp::rclcpp
+                      urdfdom_headers::urdfdom_headers)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/trajectory_processing/CMakeLists.txt b/trajectory_processing/CMakeLists.txt
index bbbe8d1f2..3a559becb 100644
--- a/trajectory_processing/CMakeLists.txt
+++ b/trajectory_processing/CMakeLists.txt
@@ -8,17 +8,11 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_trajectory_processing
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
-  moveit_trajectory_processing
-  rclcpp
-  rmw_implementation
-  urdf
-  urdfdom
-  urdfdom_headers
-  visualization_msgs
-  Boost)
 target_link_libraries(moveit_trajectory_processing moveit_robot_state
-                      moveit_robot_trajectory ruckig::ruckig)
+                      moveit_robot_trajectory ruckig::ruckig rclcpp::rclcpp
+                      ${rmw_implementation_TARGETS} urdf::urdf
+                      urdfdom_headers::urdfdom_headers
+                      ${visualization_msgs_TARGETS} ${MOVEIT_BOOST_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/transforms/CMakeLists.txt b/transforms/CMakeLists.txt
index 5880e5795..924dd48d1 100644
--- a/transforms/CMakeLists.txt
+++ b/transforms/CMakeLists.txt
@@ -6,15 +6,10 @@ target_include_directories(
 target_link_libraries(moveit_transforms moveit_macros moveit_utils)
 set_target_properties(moveit_transforms PROPERTIES VERSION
                                                    "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
-  moveit_transforms
-  geometric_shapes
-  tf2_eigen
-  rclcpp
-  rmw_implementation
-  urdfdom
-  urdfdom_headers
-  Boost)
+target_link_libraries(moveit_transforms ${geometric_shapes_TARGETS}
+                      tf2_eigen::tf2_eigen rclcpp::rclcpp
+                      ${rmw_implementation_TARGETS}
+                      urdfdom_headers::urdfdom_headers ${MOVEIT_BOOST_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/utils/CMakeLists.txt b/utils/CMakeLists.txt
index ca44183fa..73556c568 100644
--- a/utils/CMakeLists.txt
+++ b/utils/CMakeLists.txt
@@ -3,8 +3,8 @@ add_library(moveit_utils SHARED src/lexical_casts.cpp src/message_checks.cpp
 target_include_directories(
   moveit_utils PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
                       $<INSTALL_INTERFACE:include/moveit_core>)
-ament_target_dependencies(moveit_utils Boost moveit_msgs rclcpp fmt)
-target_link_libraries(moveit_utils rsl::rsl)
+target_link_libraries(moveit_utils ${MOVEIT_BOOST_TARGETS} ${moveit_msgs_TARGETS}
+                      rclcpp::rclcpp fmt::fmt rsl::rsl)
 set_target_properties(moveit_utils PROPERTIES VERSION
                                               "${${PROJECT_NAME}_VERSION}")
 
@@ -17,19 +17,15 @@ target_include_directories(
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_core>)
 target_link_libraries(moveit_test_utils moveit_robot_model
-                      moveit_kinematics_base rsl::rsl)
-ament_target_dependencies(
-  moveit_test_utils
-  ament_index_cpp
-  Boost
-  geometry_msgs
-  urdf
-  pluginlib
-  srdfdom
-  urdfdom
-  urdfdom_headers
-  rclcpp
-  fmt)
+                      moveit_kinematics_base rsl::rsl
+                      ament_index_cpp::ament_index_cpp
+                      ${MOVEIT_BOOST_TARGETS}
+                      ${geometry_msgs_TARGETS}
+                      urdf::urdf
+                      pluginlib::pluginlib
+                      ${srdfdom_TARGETS}
+                      urdfdom_headers::urdfdom_headers
+                      rclcpp::rclcpp fmt::fmt)
 set_target_properties(moveit_test_utils PROPERTIES VERSION
                                                    "${${PROJECT_NAME}_VERSION}")
 
